src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/Telemetry/ConfiguredProjectLanguageServiceTelemetryService.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System.ComponentModel.Composition;
+using Microsoft.VisualStudio.ProjectSystem;
+using Microsoft.VisualStudio.ProjectSystem.LanguageServices;
+
+namespace Microsoft.VisualStudio.Telemetry
+{
+    [Export(typeof(IConfiguredProjectLanguageServiceTelemetryService))]
+    internal class ConfiguredProjectLanguageServiceTelemetryService : IConfiguredProjectLanguageServiceTelemetryService
+    {
+        private readonly ConfiguredProject _project;
+        private readonly ITelemetryService _telemetryService;
+        private string? _projectHash;
+
+        [ImportingConstructor]
+        public ConfiguredProjectLanguageServiceTelemetryService(ConfiguredProject project, ITelemetryService telemetryService)
+        {
+            _project = project;
+            _telemetryService = telemetryService;
+        }
+
+        private string ProjectTelemetryId
+        {
+            get
+            {
+                if (Strings.IsNullOrEmpty(_projectHash))
+                {
+                    string? fullPath = _project?.UnconfiguredProject?.FullPath;
+                    _projectHash = Strings.IsNullOrEmpty(fullPath) ? string.Empty : _telemetryService.HashValue(fullPath);
+                }
+
+                return _projectHash;
+            }
+        }
+
+        public void PostDesignTimeBuildFailureEvent()
+        {
+            _telemetryService.PostProperties(TelemetryEventName.DesignTimeBuildComplete, new (string propertyName, object propertyValue)[]
+                {
+                    ( TelemetryPropertyName.DesignTimeBuildCompleteSucceeded, false),
```

Comment:
nit: spaces after opening paren here and elsewhere.

Reply:
I assumed that any spacing not changed by Edit > Advanced > Format Document was fine. Will fix.

---
In reply to: [567515572](https://github.com/dotnet/project-system/pull/6880#discussion_r567515572) [](ancestors = 567515572)

Reply:
That's fair. I guess either our code style rules are underspecified, or it's not possible to control this particular styling quirk. It's not a big deal, so don't block merging for this.